core/com.b2international.snowowl.datastore.server/src/com/b2international/snowowl/datastore/server/session/ApplicationSessionManager.java

```diff
@@ -235,6 +227,10 @@ public void loginWithResponse(final byte[] response) throws SecurityException {
 
 			final String password = new String(decryptedResponse, RANDOM_BYTES_LENGTH, decryptedResponse.length - RANDOM_BYTES_LENGTH, Charsets.UTF_8);
 			final Collection<Role> roles = authenticate(userId, password);
+			
+			if (!loginEnabled && !roles.contains(SpecialRole.ADMINISTRATOR)) {
```

Comment:
Wondering if we should ever refer to roles directly instead of permissions... 🤔 This use case would require an "administrative login" permission, though.